compiler/coreSyn/CoreLint.hs

```diff
@@ -1212,17 +1247,19 @@ lintCoreAlt :: Var              -- Scrut Var
             -> LintM UsageEnv
 -- If you edit this function, you may need to update the GHC formalism
 -- See Note [GHC Formalism]
-lintCoreAlt _ _ _ alt_ty (DEFAULT, args, rhs) =
+lintCoreAlt scrut _ _ alt_ty (DEFAULT, args, rhs) =
   do { lintL (null args) (mkDefaultArgsMsg args)
-     ; lintAltExpr rhs alt_ty }
+     ; rhs_ue <- lintAltExpr rhs alt_ty
+     ; return $ deleteUE rhs_ue scrut }
```

Comment:
How is the usage of the case-binder checked? It would seem that the same ue-alias approach for `let`s would work for the case-binder, too.